src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1581 @@
+                dropdown::populateTownSizeSelect(self, &self->widgets[widgetIndex]);
+        }
+
+        // 0x0049A844
+        static void on_resize(window* self)
+        {
+            self->set_size(windowSize, windowSize);
+        }
+
+        // 0x0049A7C7
+        static void on_close(window* self)
+        {
+            if (input::is_tool_active(self->type, self->number))
+                input::cancel_tool();
+        }
+
+        // 0x0049A3BE
+        static void tabReset(window* self)
+        {
+            self->min_width = windowSize.width;
+            self->min_height = windowSize.height;
+            self->max_width = windowSize.width;
+            self->max_width = windowSize.height;
+            self->width = windowSize.width;
+            self->height = windowSize.height;
+            input::toolSet(self, common::widx::tab_build_town, 38);
+            input::set_flag(input::input_flags::flag6);
+            common::showGridlines();
+        }
+
+        static void init_events()
+        {
+            events.draw = draw;
+            events.on_close = on_close;
+            events.on_dropdown = on_dropdown;
+            events.on_resize = on_resize;
+            events.on_tool_update = on_tool_update;
+            events.on_tool_down = on_tool_down;
+            events.on_mouse_up = on_mouse_up;
+            events.on_mouse_down = on_mouse_down;
+            events.on_update = on_update;
+            events.prepare_draw = prepare_draw;
+            events.on_tool_abort = on_tool_abort;
+        }
+    }
+
+    namespace build_buildings
+    {
+        static const gfx::ui_size_t windowSize = { 600, 172 };
+
+        static const uint8_t rowHeight = 112;
+
+        enum widx
+        {
+            scrollview = 8,
+            rotate_object,
+            object_colour,
+        };
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << scrollview) | (1 << rotate_object) | (1 << object_colour);
+
+        widget_t widgets[] = {
+            commonWidgets(639, 171, string_ids::title_build_new_buildings),
+            make_widget({ 2, 45 }, { 572, 111 }, widget_type::scrollview, 1, 2),
+            make_widget({ 575, 46 }, { 23, 23 }, widget_type::wt_9, 1, image_ids::rotate_object, string_ids::rotate_object_90),
+            make_widget({ 579, 91 }, { 15, 15 }, widget_type::wt_10, 1, image_ids::null, string_ids::tooltip_object_colour),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        // 0x0049A8A6
+        static void prepare_draw(ui::window* self)
+        {
+            if (self->widgets != widgets)
+            {
+                self->widgets = widgets;
+                self->init_scroll_widgets();
+            }
+
+            self->activated_widgets &= ~((1ULL << common::widx::tab_town_list) | (1ULL << common::widx::tab_build_town) | (1ULL << common::widx::tab_build_buildings) | (1ULL << common::widx::tab_build_misc_buildings));
```

Comment:
This is taken care of by common::prepareDraw. Can you go through and remove the common sections.